src/FieldControl.js

```diff
@@ -3,32 +3,38 @@ import PropTypes from 'prop-types'
 import { FormControl, FormArray, FormGroup } from './model'
 import configureControl from './configureControl'
 import Field from './Field'
-export default class FieldControl extends React.Component {
-  constructor(props, context) {
-    super(props, context)
-    this.control = configureControl(props, context, 'FormControl')
-  }
-  componentWillReceiveProps(nextProps) {
-    const { name } = nextProps
-    if (this.props.name !== name) {
-      this.control = configureControl(nextProps, this.context, 'FormControl')
+
+class FieldControlInternal extends React.Component {
+  state = {}
+
+  static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
Same here, just put the `componentWillReceiveProps` logic in `componentDidUpdate`.
There is no need to define the `state`.

Reply:
That makes my original approach look rather silly :) Thanks, this should be addressed in #54